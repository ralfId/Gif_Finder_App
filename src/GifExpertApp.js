import React, {useState} from 'react';
import { AddCategory } from './components/AddCategory';
import { GridCategory } from './components/GridCategory';

export const GifExpertApp = ({defaultCategories = []}) => {

    //const catgList = ['one pice', 'hxh', 'Naruto']
    const [category, setCategory] = useState(defaultCategories)
    
    // const handleAdd = ()=>{
    //     /* Correct ways to push an element */
    //     // setCategory(['7 Deadly Sign', ...category])
    //     setCategory( categ => ['one punch man', ...category] );
    // }


    return (
        <>
            <h2>Gif Expert App</h2>
            <AddCategory setCategory= { setCategory }/>
            <hr />
            <ol>
            {
                category.map(ctg => 
                    <GridCategory key= {ctg} category={ctg} />
                )
            }
            </ol>
            

        </>
    )
};
