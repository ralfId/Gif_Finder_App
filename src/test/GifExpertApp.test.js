import { shallow } from "enzyme";
import { GifExpertApp } from "../GifExpertApp";

describe('Pruebas en GifExpertApp.js', () => {
    
    test('Debe hacer match con el snapshop de <GifExpertApp/>', () => {
        
        const wrapper = shallow(<GifExpertApp />);
        expect(wrapper).toMatchSnapshot();
    });


    test('Debe verificar si <GridCategory/> existe n veces', () => {
        
        const categories = ['One piece', 'xHx']
        const wrapper = shallow(<GifExpertApp defaultCategories={categories}/>);

        expect(wrapper).toMatchSnapshot();
        expect(wrapper.find('GridCategory').length).toBe(categories.length);
    });
});