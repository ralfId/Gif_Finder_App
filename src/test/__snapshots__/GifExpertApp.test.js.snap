// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en GifExpertApp.js Debe hacer match con el snapshop de <GifExpertApp/> 1`] = `
<Fragment>
  <h2>
    Gif Expert App
  </h2>
  <AddCategory
    setCategory={[Function]}
  />
  <hr />
  <ol />
</Fragment>
`;

exports[`Pruebas en GifExpertApp.js Debe verificar si <GridCategory/> existe n veces 1`] = `
<Fragment>
  <h2>
    Gif Expert App
  </h2>
  <AddCategory
    setCategory={[Function]}
  />
  <hr />
  <ol>
    <GridCategory
      category="One piece"
      key="One piece"
    />
    <GridCategory
      category="xHx"
      key="xHx"
    />
  </ol>
</Fragment>
`;
